x/staking/side_handler.go

```diff
@@ -86,7 +86,9 @@ func SideHandleMsgValidatorJoin(ctx sdk.Context, msg types.MsgValidatorJoin, k k
 	}
 
 	// decode validator join event
-	eventLog, err := contractCaller.DecodeValidatorJoinEvent(chainParams.StakingInfoAddress, receipt, msg.LogIndex)
+	var stakingInfoAddress [20]byte
+	copy(stakingInfoAddress[:], chainParams.StakingInfoAddress)
+	eventLog, err := contractCaller.DecodeValidatorJoinEvent(stakingInfoAddress, receipt, msg.LogIndex)
```

Comment:
this looks hacky, any better way to handle this?

Reply:
@jdkanani

Reply:
We have to decode events and validator them in the side handler. Not sure how to get rid of it.

Reply:
@murthyvitwit for now let's create a util function and use it from there

Reply:
Sorry for the misunderstanding. We can avoid conversion.

Reply:
removed it